### PR TITLE
fix(docs): Update retina docs to include Linux capabilities used by each plugin

### PR DESCRIPTION
**Pull request by BeegiiK**
# Description

Update retina documentation to include Linux capabilities used by each plugin. My original report is [here](https://microsoft.sharepoint.com/:o:/r/teams/Aznet/_layouts/15/Doc.aspx?sourcedoc=%7B29805a19-9c3d-4ada-bc99-f1f233f74345%7D&action=edit&wd=target(Retina.one%7C29ecdb42-0a8e-4808-8fa7-843a2984373f%2FEvaluating%20Security%20Context%20%5C%2F%20Capabilities%7C690e30cf-b717-4d90-981b-01459aae6f19%2F)&wdorigin=703&wdpreservelink=1&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNDA5MTIyMTMxOSJ9&share=IgEZWoApPZzaSryZ8fIz90NFAa1Bg4o7KybtXMEWmCqVv2w). Not all plugins could be evaluated and is mentioned in my report.

I've confirmed that the minimum required set of capabilities to run all the plugins (except `ciliumeventobserver` due to a failure) through cluster testing, are the following: 
- `SYS_ADMIN`
- `NET_ADMIN` 
- `IPC_LOCK `

## Related Issue

https://github.com/microsoft/retina/issues/93

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: docs/03-Metrics/plugins/readme.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Each metric is associated with a Plugin.
 Associated metrics are linked below.
 See [Metrics Configuration](../configuration.md) for info on configuration.
 
+To run Retina without any plugins, the `CAP_BPF` Linux capability is required for memory locking and for loading/using BPF programs. This capability is mandatory. If you're using any plugins, ensure that the necessary capabilities for those plugins are also added. If a plugin requires `CAP_SYS_ADMIN`, you can substitute it for `CAP_BPF`.
```
Comment (nddq) [inline docs/03-Metrics/plugins/readme.md:7]: > If a plugin requires `CAP_SYS_ADMIN`, you can substitute it for `CAP_BPF`.
Does this actually work? Last time I tested this it didn't 🥲

Comment (BeegiiK) [reply docs/03-Metrics/plugins/readme.md:7]: Yes @nddq! I created the retina binary and ran it with different capabilities with and without plugins

Comment (nddq) [reply docs/03-Metrics/plugins/readme.md:7]: Can you make a change to the helm deployment files to test this out in our pipeline? Also, I remember `CAP_BPF` is only available from a specific kernel version onward, so we should clarify it as well.

**Review by anubhabMajumdar — APPROVED** [review]

File: docs/03-Metrics/plugins/readme.md (hunk 1 above)
Comment (BeegiiK) [reply docs/03-Metrics/plugins/readme.md]: Done! Updated in latest commit

**Review by anubhabMajumdar — APPROVED** [review]

**Review by nddq — APPROVED** [review]

**Pull request merged**